src/entities/marking/constants/queryKey.ts

```diff
@@ -0,0 +1,65 @@
+interface LatLng {
+  lat: number;
+  lng: number;
+}
+
+interface Bounds {
+  southWestLat: number;
+  southWestLng: number;
+  northEastLat: number;
+  northEastLng: number;
+}
+
+type SortType = "RECENT" | "DISTANCE" | "POPULARITY";
+type SearchType = "NEARBY" | "LOCATION";
+
+type Activity = "LIKED" | "SAVED";
+
+export const markingQueryKey = {
+  address: (latLng: LatLng) => ["address", latLng] as const,
+
+  markerAll: () => ["marker"] as const,
+  markingListAll: () => ["marking"] as const,
+
+  boundaryMarker: (bounds: Bounds) =>
+    [...markingQueryKey.markerAll(), { bounds }] as const,
+  myActivityMarker: (activity: Activity) =>
+    [...markingQueryKey.markerAll(), { activity }] as const,
+  myMarker: () => [...markingQueryKey.markerAll(), "myMarker"] as const,
+  markerThumbnail: (nickname: string) =>
+    [...markingQueryKey.markerAll(), "dashboard", { nickname }] as const,
+
+  detail: (markingId: number) =>
+    [...markingQueryKey.markingListAll(), { markingId }] as const,
+
+  boundaryMarkingList: (
+    bounds: Bounds,
+    latLng: LatLng,
+    sortType: SortType,
+    searchType: SearchType,
+  ) =>
+    [
+      ...markingQueryKey.markingListAll(),
+      { bounds },
+      { latLng },
+      { sortType },
+      { searchType },
+    ] as const,
```

Comment:
```suggestion
    [
      ...markingQueryKey.markingListAll(),
      bounds,
      latLng,
      sortType,
      searchType,
    ] as const,
```

인수를 객체로 감싸고 있는데, bounds랑 latLng은 지워도 될 것 같아요.
그리고 sortType과 searchType은 string인데 객체로 관리하는 이유가 있을까요?

<img width="1044" alt="스크린샷 2024-11-13 오후 5 51 13" src="https://github.com/user-attachments/assets/fb8e8e9f-178d-405e-9061-570f358584f6">

Reply:
객체로 관리하면 키값을 통해 해당 값이 어떤 의미를 가진 변수인지를 나타낼 수 있어서 저는 객체로 관리하는것을 선호하는데요 

예를 들어 `bounds` 객체 같은 경우도 

`{south  : .... , north : ...}` 이렇게 좌표값만 띠롱 있는 편보다 `{bounds : {south : .. , noth : ..}}` 이런식으로 있는 편을 선호합니다.

내부에 어떤 값이 무슨 의미로 있는지 키값을 통해 보이니까요 

또 `string` 값들도 어떤 의미를 갖느 쿼리키인지 드러나는 편이 보기 좋지 않나요 ? 

만약 닉네임, 소트 타입 , 서치타입이 쿼리키로 있을 때 

`string , string , string` 으로 있는 편 보다 `{nickname : string} , {sortType : string} .. ` 이런편이 더 좋아보여서요

Reply:
그 객체에 키가 bounds 하나만 있을 거라면, `{south  : .... , north : ...}` 형태로 있어도 키 네이밍들이 bounds라는 의미를 나타낸다고 생각했습니다.
sortType이나 searchType은 둘 다 선택한 옵션이라 `{ sortType, searchType }`처럼 하나의 객체로 둬도 괜찮을 것 같은데 따로 두신 이유가 있나요?

Reply:
저는 bounds 가 하나만 있더라도 `bounds  :  {}` 형태로 있는편이 나쁘지 않다고 생각이 들어요 

JSON.stringfy 를 통해 허용 되는 직렬화 수준이기도 하고 , 네이밍 만으로 bounds 만을 알 수도 있겠지만 있다고 해서 도움이 되면 되었지 안좋아 보이지는 않아서요 

저거 두 개를 따로 둘지 같이 둘지는 혹시 나중에 invalidate 작업 할 때 변경해도 될까요 , 지금은 저렇게 두나 같이 두나 문제 없어 보이는데 나중에 invalidate하거나 다른 캐싱 최적화 작업 할 때 문제가 드러나면 그 떄 건들여야 할 거 같아요

Reply:
현재는 팩토리 패턴으로 만든 쿼리키에서 해당 객체의 메소드들을 실제 다른 쿼리나 뮤테이션에서 사용 할 때 타입을 변경할 가능성이 많습니다. 그 때 가봐야 쿼리키를 어떻게 다시 바꿀지 정확하게 감이 올 거 같아요 

그래서 현재의 상태는 완벽하지 못한 점 유념해주시면 좋겠습니다